test/position-managers/PositionManager.modifyLiquidities.t.sol

```diff
@@ -308,4 +336,353 @@ contract PositionManagerModifyLiquiditiesTest is Test, PosmTestSetup, LiquidityF
+        planner.add(Actions.SETTLE, abi.encode(address(_WETH9), ActionConstants.OPEN_DELTA, false));
+        // other currency can close normally
+        planner.add(Actions.CLOSE_CURRENCY, abi.encode(currency1));
+        // we wrapped the full contract balance so we sweep back in the wrapped currency
+        planner.add(Actions.SWEEP, abi.encode(CurrencyLibrary.ADDRESS_ZERO, ActionConstants.MSG_SENDER));
+        bytes memory actions = planner.encode();
+
+        lpm.modifyLiquidities{value: 102 ether}(actions, _deadline);
+
+        uint256 balanceEthAfter = address(this).balance;
+        uint256 balance1After = IERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+
+        // Approx 100 eth was spent because the extra 2 were refunded.
+        assertApproxEqAbs(balanceEthBefore - balanceEthAfter, 100 ether, 1 wei);
+        assertApproxEqAbs(balance1Before - balance1After, 100 ether, 1 wei);
+        assertEq(lpm.ownerOf(tokenId), address(this));
+        assertEq(lpm.getPositionLiquidity(tokenId), liquidityAmount);
+        assertEq(_WETH9.balanceOf(address(lpm)), 0);
+        assertEq(address(lpm).balance, 0);
+    }
+
+    function test_wrap_increaseLiquidity_usingExactAmount() public {
+        // weth-currency1 pool initialized as wethKey
+        // input: eth, currency1
+        // modifyLiquidities call to mint liquidity weth and currency1
+        // 1 _wrap with an amount
+        // 2 _mint
+        // 3 _settle weth where the payer is the contract
+        // 4 _close currency1, payer is caller
+        // 5 _sweep weth since eth was entirely wrapped
+
+        uint256 balanceEthBefore = address(this).balance;
+        uint256 balance1Before = IERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+        uint256 tokenId = lpm.nextTokenId();
+
+        uint128 liquidityAmount = LiquidityAmounts.getLiquidityForAmounts(
+            SQRT_PRICE_1_1,
+            TickMath.getSqrtPriceAtTick(wethConfig.tickLower),
+            TickMath.getSqrtPriceAtTick(wethConfig.tickUpper),
+            100 ether,
+            100 ether
+        );
+
+        Plan memory planner = Planner.init();
+        planner.add(Actions.WRAP, abi.encode(100 ether));
+        planner.add(
+            Actions.MINT_POSITION,
+            abi.encode(
+                wethConfig.poolKey,
+                wethConfig.tickLower,
+                wethConfig.tickUpper,
+                liquidityAmount,
+                MAX_SLIPPAGE_INCREASE,
+                MAX_SLIPPAGE_INCREASE,
+                ActionConstants.MSG_SENDER,
+                ZERO_BYTES
+            )
+        );
+
+        // weth9 payer is the contract
+        planner.add(Actions.SETTLE, abi.encode(address(_WETH9), ActionConstants.OPEN_DELTA, false));
+        // other currency can close normally
+        planner.add(Actions.CLOSE_CURRENCY, abi.encode(currency1));
+        // we wrapped the full contract balance so we sweep back in the wrapped currency for safety measure
```

Comment:
again not true 100 ETH was wrapped not CONTRACT_BALANCE

Reply:
We didnt use CONTRACT_BALANCE but 100 eth IS the full contract balance so we will are sweeping back in weth